### ISSUE TITLE
Hyperparameters to reproduce the result of qwen2-1.5B model on ml-1m?

### ISSUE THREAD

**Issue by bebigger** [zyang1580/CoLLM#23]
你好。我在以qwen2-1.5B为backbone复现时遇到了问题。
一开始我使用两张3090进行实验，将batch_size调整至8，并降低学习率。训练好LORA模块后，模型的效果与论文中的Tallrec模型相差不多，但是在训练完CIE模块后，最终结果大幅度下降。
后面我使用两张A800（80G）进行训练，使用最初的超参数，以下是我训练LORA模块后测试的结果：
> 2025-06-07 16:22:38,193 [INFO] Averaged stats: loss: 0.629959  acc: 0.267258 ***auc: 0.7022502138301632 ***uauc:0.6839774080629103
rank_0 auc: 0.6956601671309193

然后我使用预训练好的mf模型（测试效果与论文里的结果）训练CIE模块，最终的测试结果还是大幅度下降
> 2025-06-07 21:52:01,431 [INFO] Averaged stats: loss: 5.306555  acc: 0.570495 ***auc: 0.5786041566872622 ***uauc:0.60812455855067
rank_0 auc: 0.5905763516568697

请问在训练CIE模块时是否需要额外调整超参数，或者是其他什么我没注意到的因素导致了这种差距？

以下是预训练的mf模型的测试结果：

> valid_auc:0.6788422896403246, valid_uauc:0.6269208527178279, test_auc:0.7273190725748894, test_uauc:0.6627275604526212, acc: 0.5897509854821652

以下是我训练时的参数设置：

> =====  Running Parameters    =====
2025-06-07 21:50:15,251 [INFO] {
    "amp": true,
    "batch_size_eval": 64,
    "batch_size_train": 48,
    "device": "cuda",
    "dist_backend": "nccl",
    "dist_url": "env://",
    "distributed": true,
    "evaluate": true,
    "gpu": 0,
    "init_lr": 0.001,
    "iters_per_epoch": 50,
    "lr_sched": "linear_warmup_cosine_lr",
    "max_epoch": 1000,
    "min_lr": 8e-05,
    "mode": "v2",
    "num_workers": 4,
    "output_dir": "/root/autodl-tmp/CoLLM-main/Qwen/Qwen_finetune",
    "rank": 0,
    "resume_ckpt_path": null,
    "seed": 42,
    "task": "rec_pretrain",
    "test_splits": [
        "test",
        "valid",
        "test_warm",
        "test_cold"
    ],
    "train_splits": [
        "train"
    ],
    "valid_splits": [
        "valid"
    ],
    "warmup_lr": 1e-05,
    "warmup_steps": 200,
    "weight_decay": 0.001,
    "world_size": 2
}
2025-06-07 21:50:15,251 [INFO] 
======  Dataset Attributes  ======
2025-06-07 21:50:15,251 [INFO] 
======== amazon_ood =======
2025-06-07 21:50:15,251 [INFO] {
    "build_info": {
        "storage": "collm-datasets/ml-1m/"
    },
    "data_type": "default",
    "path": "collm-datasets/ml-1m/"
}
2025-06-07 21:50:15,251 [INFO] 
======  Model Attributes  ======
2025-06-07 21:50:15,251 [INFO] {
    "ans_type": "v2",
    "arch": "mini_gpt4rec_v2",
    "ckpt": "/root/autodl-tmp/CoLLM-main/Qwen/Qwen_finetune/20250607211/checkpoint_best.pth",
    "end_sym": "###",
    "freeze_lora": true,
    "freeze_proj": false,
    "freeze_rec": true,
    "item_num": -100,
    "llama_model": "Qwen/Qwen2-1.5B",
    "lora_config": {
        "alpha": 16,
        "dropout": 0.05,
        "r": 8,
        "target_modules": [
            "q_proj",
            "v_proj"
        ],
        "use_lora": true
    },
    "max_txt_len": 1024,
    "model_type": "pretrain_vicuna",
    "proj_drop": 0,
    "proj_mid_times": 10,
    "proj_token_num": 1,
    "prompt_path": "prompts/collm_movie.txt",
    "prompt_template": "{}",
    "rec_config": {
        "embedding_size": 256,
        "item_num": 3256,
        "pretrained_path": "PretrainedModels/mf/0912_ml1m_oodv2_best_model_d256lr-0.001wd0.0001.pth",
        "user_num": 839
    },
    "user_num": -100
}

**Comment by zyang1580** [comment]
印象中主要是调整下学习率，你可以调整试试。

**Comment by qiu58** [comment]
@bebigger 你好，可以请教下Qwen2-1.5B如何复现吗？

**Comment by Shengxiang-Lin** [comment]
@bebigger 您好，我也遇到了和您相似的问题，请问您后来是如何解决的？
